libavformat/mpjpegdec.c

```diff
@@ -282,6 +285,24 @@ static char* mpjpeg_get_boundary(AVIOContext* pb)
     return res;
 }
 
+static int shrink_buffer(AVIOContext *pb) {
+    const int data_len =  pb->buf_end - pb->buf_ptr;
+    const int new_buffer_size = data_len;
+    uint8_t *orig_buffer = pb->buffer;
+    uint8_t *new_buffer = NULL;
+
```

Comment:
One line is missing from https://github.com/camect/FFmpeg/commit/e6b947f7c6db8dbb78191d581c89d317f0053737#diff-07eb4b9eedb94df49c6f9d0bc8c707877066eeac68d18ac3ef532d334c200145

```
if (pb->buffer_size == pb->orig_buffer_size) return 0;
```

Reply:
Done.

Reply:
Right, I have done like this because I could not get the orig_buffer_size any more, you could see I change the line 290 either.
I'm thinking convert the `uint8_t *orig_buffer = pb->buffer;` to string and get the length by myself:
`int length = std::strlen(reinterpret_cast<const char*>(orig_buffer));`

Do you think it's appropriate?

Reply:
no. the buffer is not a string..
you can simply shrink buffer to data_len

Reply:
Done.
I'll leave it as-is.